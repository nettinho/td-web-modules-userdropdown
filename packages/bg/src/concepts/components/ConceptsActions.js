import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Button } from "semantic-ui-react";
import { Link } from "react-router-dom";
import { bindActionCreators } from "redux";
import { FormattedMessage } from "react-intl";
import { downloadConcepts, uploadConcepts } from "../routines";
import routes from "../routes";
import ConceptsUploadButton from "./ConceptsUploadButton";

export const ConceptsActions = ({
  createUrl,

  conceptsDownloading,
  downloadConcepts
}) => (
  <div>
    <ConceptsUploadButton />
    {createUrl && (
      <Button
        style={{ float: "right" }}
        secondary
        as={Link}
        to={createUrl}
        content={<FormattedMessage id="concepts.actions.create" />}
      />
    )}
    <Button
      style={{ float: "right" }}
      secondary
      onClick={() => downloadConcepts()}
      loading={conceptsDownloading}
      content={<FormattedMessage id="concepts.actions.download" />}
    />
  </div>
);

ConceptsActions.propTypes = {
  conceptsDownloading: PropTypes.bool,
  createUrl: PropTypes.string,
  downloadConcepts: PropTypes.func
};

const mapStateToProps = ({
  conceptLoading,
  conceptActions,
  conceptsActions,
  conceptsDownloading
}) => ({
  createUrl:
    !conceptLoading &&
    _.isEmpty(conceptActions) &&
    _.has("create")(conceptsActions)
      ? routes.CONCEPTS_NEW
      : undefined,
  conceptsDownloading
});

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ downloadConcepts, uploadConcepts }, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ConceptsActions);
