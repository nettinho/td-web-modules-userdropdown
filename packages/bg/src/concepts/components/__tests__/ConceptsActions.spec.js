import React from "react";
import { shallow } from "enzyme";
import { ConceptsActions } from "../ConceptsActions";

describe("<ConceptsActions />", () => {
  const props = { createUrl: "/url/to/create/concept" };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<ConceptsActions {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
