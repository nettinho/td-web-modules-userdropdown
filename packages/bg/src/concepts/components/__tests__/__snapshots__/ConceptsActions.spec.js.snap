// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConceptsActions /> matches the latest snapshot 1`] = `
<div>
  <Connect(Component) />
  <Button
    as={[Function]}
    content={
      <FormattedMessage
        id="concepts.actions.create"
        values={Object {}}
      />
    }
    role="button"
    secondary={true}
    style={
      Object {
        "float": "right",
      }
    }
    to="/url/to/create/concept"
  />
  <Button
    as="button"
    content={
      <FormattedMessage
        id="concepts.actions.download"
        values={Object {}}
      />
    }
    onClick={[Function]}
    role="button"
    secondary={true}
    style={
      Object {
        "float": "right",
      }
    }
  />
</div>
`;
